crates/zksync/core/src/vm/runner.rs

```diff
@@ -189,7 +196,7 @@ where
         Some(factory_deps),
         PaymasterParams::default(),
     );
-    inspect(tx, env, db, journaled_state, ccx)
+    inspect(tx, env, db, journaled_state, ccx, call.caller)
```

Comment:
Shouldn't this be the retrieved env.tx.caller? Or rather, why isn't it?

Reply:
the last parameter is the `msg.sender`'s value. `env.tx.caller` always contains the account that initiated (and paid for the tx), meanwhile `msg.sender` changes when Contracts make call as well. Hence this value is supposed to represent the deepest caller context.